platform/darwin/src/MGLShapeSource.h

```diff
@@ -305,6 +307,10 @@ MGL_EXPORT
  */
 - (NSArray<id <MGLFeature>> *)featuresMatchingPredicate:(nullable NSPredicate *)predicate;
 
+// TODO: doc
```

Comment:
When you do write the docs for this part, I think its a good idea to add extra clarification on what are leaves vs. clusters since the metaphor is a bit odd..